### PR TITLE
Gists: publish a note to a secret GitHub Gist from Note actions

### PR DESCRIPTION
**Pull request by maccman**
## What

Adds **Publish to gist**: a note's body can be published to a **secret GitHub Gist** from the Note actions section (next to Pin / Lock) or the ⌘K palette. Publishing copies the gist link to the clipboard; after the first publish the button flips to **Republish gist**, which updates the **same** gist in place. An accent-tinted icon + tooltip nudge when the body changed since the last publish.

## How

- **Frontmatter is the record** (`gist:` block — id, url, file, hash), following the pinned/private pattern: written through the live session's `commitFrontmatter` when the note is open, read-patch-write on disk otherwise.
- **Staleness is a body-hash comparison**, not a timestamp — publishing writes the `gist` block itself, so mtime-based detection would flag every note the instant it was published. The hash (sync FNV-1a, body only) is computed at publish time and re-computed at index time; frontmatter-only changes (pin, private) never nudge a republish.
- **Projection**: migration `0009_gist.sql` adds `notes.gist_url` + `notes.gist_stale` (projection-wipe pattern), `PROJECTION_VERSION` → 5, Rust `IndexedNote` mirror + Kysely codegen updated.
- **Republish addresses the file by its previous name** (`PATCH /gists/{id}` keys files by current name), so title changes rename instead of adding a second file — and files a user added to the gist by hand are never touched. A gist deleted on github.com falls back to creating a fresh one.
- **`private: true` stays a hard block**: the action hides for private notes, and the publish path runs `assertCloudAllowed` on the live content before any byte leaves the device. `PrivateNoteError`'s message generalizes "AI service" → "external service" to match the actual contract.
- The action renders only when a GitHub credential is stored (`useGithubConnected`; queries cache forever, so every connect/disconnect flow now signals `invalidateGithubAuth`).

## ⚠️ Action needed on the GitHub App

Gist endpoints work with the existing device-flow user tokens, **but `reflect-github-app` needs the "Gists" account permission (read & write)** added under *Permissions & events → Account permissions* on github.com, and users must re-run the device flow once to mint a token carrying it (fine since the app isn't live). Without it GitHub deliberately answers **404**, which `createGist` maps to a "reconnect GitHub to grant it" auth error. Fine-grained PATs need the same Gists account permission on the token.

## Tests

- Core: gists API (create/update/rename/404 fallback/auth mapping), body hash + filename, nested-mapping frontmatter round-trip + coercion edge cases, projection staleness.
- Desktop: publish flow (session routing, republish, deleted-gist fallback, privacy/empty/no-token refusals, clipboard + operations feedback), action component visibility/label bridge.
- Rust: gist column round-trip, migration version bumps.
- `pnpm check` green; core 507 + desktop suites + `cargo test` (db, index-schema, cli) pass.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Medium Risk**
> Touches external data export (gist content), auth/token use, and a projection-wipe migration that forces note re-indexing on upgrade; privacy and compensation logic are explicitly tested but gist permission misconfiguration affects users at runtime.
> 
> **Overview**
> Adds **Publish to gist** / **Republish gist** from Note actions and the command palette (`note.publishGist`), gated on stored GitHub credentials and hidden for private notes.
> 
> Publishing sends the note body to a **secret** gist via new core GitHub Gist APIs (`createGist` / `updateGist` / `deleteGist`), records a `gist` frontmatter block (id, url, file, body hash), copies the link to the clipboard, and republishes to the **same** gist when the block exists. Staleness for the republish nudge is a **body-only hash** (not mtime), projected as `gist_url` / `gist_stale` in SQLite migration **0009** and `PROJECTION_VERSION` 5, with matching Rust index writes and `NoteRow` fields.
> 
> The publish path respects live session `commitFrontmatter`, blocks `private: true` and invalid frontmatter before network I/O, and compensates by deleting only **newly created** gists if local recording fails. `invalidateGithubAuth` keeps `useGithubConnected` in sync after sign-in/out.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 01baf8b317fc677c5351e93949eac1c81dafc5d5. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: apps/desktop/src/components/context-sidebar/note-gist-action.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  // (url landed and staleness recomputed) or the action moved to another note.
+  if (
+    pending !== null &&
+    (pending.path !== path || (row !== null && row.gistUrl === pending.url && !row.gistStale))
+  ) {
+    setPending(null)
+  }
+  const bridged = pending !== null && pending.path === path
+  const published = bridged || (row?.gistUrl ?? null) !== null
+  const stale = !bridged && (row?.gistStale ?? false)
```
[the remark this answers is not in the document]
Comment (maccman) [reply apps/desktop/src/components/context-sidebar/note-gist-action.tsx:56]: Fixed in 364108e — the bridge now retires on url match alone (waiting for gistStale to also clear could hold it forever and mute the nudge); added a regression test for the same-url republish path.

File: apps/desktop/src/lib/note-gist.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  try {
+    const url = await publishNoteToGist(path, generation)
+    await navigator.clipboard.writeText(url)
+    operation.done()
+    startOperation('Gist link copied').done()
+    return url
+  } catch (cause) {
+    operation.fail(errorMessage(cause))
+    return null
+  }
```
[the remark this answers is not in the document]
Comment (maccman) [reply apps/desktop/src/lib/note-gist.ts:118]: Fixed in 364108e — publish and clipboard copy now have separate operation scopes: a failed copy surfaces its own failure line and still returns the url, so the publish never reads as failed.

File: apps/desktop/src/lib/note-gist.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    id: published.id,
+    url: published.htmlUrl,
+    file: filename,
+    hash: gistBodyHash(body),
+  }
+  if (owner === null || !(await owner.commitFrontmatter({ gist }))) {
+    const onDisk = await readNoteOrEmpty(path)
+    await writeNote(path, upsertFrontmatter(onDisk, { gist }), generation)
+  }
+  return published.htmlUrl
```
[the remark this answers is not in the document]
Comment (maccman) [reply apps/desktop/src/lib/note-gist.ts:97]: Fixed in 01baf8b — publish now fails fast on invalid frontmatter (the likeliest record-write refusal) before anything reaches GitHub, and a freshly created gist is best-effort DELETEd when its local record fails, so the next publish can't fork an orphan. A failed republish record keeps the old block (same gist next time) and never deletes the shared gist. One nuance: the failure was already surfaced (the record write rejects through the operation line); the orphan was the real gap.
